hs_core/management/commands/check_unknown_funder_names_published_resources.py

```diff
@@ -0,0 +1,93 @@
+import requests
+from django.conf import settings
+from django.core.exceptions import ObjectDoesNotExist
+from django.core.management.base import BaseCommand
+
+from hs_core.hydroshare.utils import get_resource_by_shortkey
+from hs_core.models import BaseResource
+
+
+class Command(BaseCommand):
+    help = "List unknown funder names (names that are not in crossref funders registry for published resource(s)"
+
+    def add_arguments(self, parser):
+
+        # # a list of resource id's, or none to check all published resources
+        parser.add_argument('resource_ids', nargs='*', type=str)
+
+    def handle(self, *args, **options):
+        requests.packages.urllib3.disable_warnings()    # turn off SSL warnings
+
+        def check_funder_name(funder_name):
+            """Checks if the funder name exits in Crossref funders registry.
+            Crossref API Documentation: https://api.crossref.org/swagger-ui/index.html#/Funders/get_funders
+            """
+
+            # match all words in the funder name
+            query = "+".join(funder_name.split())
+            # if we can't find a match in first 50 search records then we are not going to find a match
+            max_record_count = 50
+            email = settings.DEFAULT_SUPPORT_EMAIL
+            url = f"https://api.crossref.org/funders?query={query}&rows={max_record_count}&mailto={email}"
+            funder_name = funder_name.lower()
+            response = requests.get(url, verify=False)
+            found_match = False
+            if response.status_code == 200:
+                response_json = response.json()
+                if response_json['status'] == 'ok':
+                    items = response_json['message']['items']
+                    for item in items:
+                        if item['name'].lower() == funder_name:
+                            found_match = True
+                            return found_match
+                        for alt_name in item['alt-names']:
+                            if alt_name.lower() == funder_name:
+                                found_match = True
+                                return found_match
+                    return found_match
+                return found_match
```

Comment:
maybe this function could return "near" matches, if there is no exact match?
That way we could include near matches in the notifications to owners. I'm thinking something like:

> " Dear owner of resource, your funder name doesn't match known funders. Some approximate matches in the registry of funders include __fill near matches here___"

Reply:
this is probably unnecessary, sorry I was just brainstorming